### PR TITLE
Feature/moveit

### PR DESCRIPTION
**Pull request by TimBuckers**
Trajectory controller with moveit 
- [x] Basic trajectory controller from ROS control
- [x] Trajectory controller linked to gazebo for the simulation
- [x] Moveit is able to create a trajectory from current position to targeted pose
- [x] Moveit is connected with gazebo to obtain the actual/current position of the exoskeleton
closes project-march/march-iv#125 Gait processing

### CODE REVIEW

File: march_control/src/control_node/ControlNode.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      return true;
+    case actionlib_msgs::GoalStatus::RECALLED:
+      return true;
+  }
+  return false;
+}
+
+void overrideTolerances(control_msgs::FollowJointTrajectoryActionGoal* trajectoryMsg)
+{
+  std::string joint_names[] = { "left_hip", "left_knee", "left_ankle", "right_hip", "right_knee", "right_ankle" };
```
Comment (Ishadijcks) [inline march_control/src/control_node/ControlNode.cpp]: Hardcoded, maybe you can ask moveit which controllers are active?

File: march_description/urdf/march-iii.xacro [hunk 2]
```diff
@@ -18,18 +18,18 @@
...
-    <xacro:property name="knee_rotation_upper_limit" value="${90*pi/180}"/>
+    <xacro:property name="knee_rotation_lower_limit" value="${-5*pi/180}"/>
+    <xacro:property name="knee_rotation_upper_limit" value="${100*pi/180}"/>
     <xacro:property name="knee_effort_limit" value="1000.0"/>
     <xacro:property name="knee_velocity_limit" value="1000.0"/>
 
-    <xacro:property name="ankle_rotation_lower_limit" value="${-50*pi/180}"/>
-    <xacro:property name="ankle_rotation_upper_limit" value="${30*pi/180}"/>
+    <xacro:property name="ankle_rotation_lower_limit" value="${-15*pi/180}"/>
+    <xacro:property name="ankle_rotation_upper_limit" value="${20*pi/180}"/>
```
Comment (Ishadijcks) [inline march_description/urdf/march-iii.xacro]: Change values to correct m3 values

File: march_moveit/config/march.srdf [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        <joint name="right_knee" value="0" />
+    </group_state>
+    <group_state name="sit" group="full_exo">
+        <joint name="left_ankle" value="0" />
+        <joint name="left_hip" value="1.4034" />
+        <joint name="left_knee" value="1.4231" />
+        <joint name="right_ankle" value="0" />
+        <joint name="right_hip" value="1.4034" />
+        <joint name="right_knee" value="1.4231" />
+    </group_state>
```
Comment (Ishadijcks) [inline march_moveit/config/march.srdf]: Remove unnecessary group_states not created for full_exo

File: march_control/config/config.yaml [hunk 4]
```diff
@@ -1,38 +1,49 @@
...
-    pid: {p: 20, i: 0.019989, d: 25}
-
-  right_ankle_position_controller:
-    type: effort_controllers/JointPositionController
-    joint: right_ankle
-    pid: {p: 450, i: 1.7, d: 12}
\ No newline at end of file
+    state_publish_rate:  25            # Override default
+    action_monitor_rate: 30            # Override default
+    stop_trajectory_duration: 0        # Override default
```
[the remark this answers is not in the document]
Comment (TimBuckers) [reply march_control/config/config.yaml]: The visualization works fine and for now a higher rate is only making it harder to run on your machine.

**Pull request merged**